lib/matplotlib/axes/_base.py

```diff
@@ -1635,16 +1635,28 @@ def relim(self, visible_only=False):
         # Collections are deliberately not supported (yet); see
         # the TODO note in artists.py.
         self.dataLim.ignore(True)
-        self.dataLim.set_points(mtransforms.Bbox.null().get_points())
-        self.ignore_existing_data_limits = True
+        # check if there is a need to create a null bbox
+        if (self.conditionCheck()):
```

Comment:
I don't see a need to define a separate function here. I would just make if the simple `if not self.lines and not self.patches:`